### PR TITLE
Bugfix - Prevent PHP Warning for count on null on LDAP

### PR DESCRIPTION
**Pull request by lippl**
When initializing the `$pwdGraceUseTimeCount` it is not checked if the `$pwdGraceUseTime` may be `null`, when pwdgraceusetime attribute my be not defined in LDAP. Starting PHP 7.2 count throws a warning when called on `null`. 
`[PHP 7.2] count(): Parameter must be an array or an object that implements Countable`
As the ldap search also only returns arrays, `[]` is used instead of `null`.
The redundant check for the variable in the if query is also removed, as only arryas, which implement countable, can be passed.
Additionally this is harmonized for all other LDAP attributes querried within the `handlePasswordExpiry` function.

### CODE REVIEW

**Comment by georgehrke** [comment]
@lippl Is this related to any existing issue?

**Comment by georgehrke** [comment]
@lippl Can you please add unit tests for it? Thx!

**Review by blizzz — APPROVED** [review]

**Comment by blizzz** [comment]
@lippl looks good from my point of view, just perhaps you can stash the commits into one?

File: apps/user_ldap/lib/User/User.php [hunk 1]
```diff
@@ -752,16 +751,16 @@ public function handlePasswordExpiry($params) {
...
 				'user_ldap.renewPassword.showRenewPasswordForm', array('user' => $uid)));
 				exit();
 			}
 			//handle password expiry warning
-			if($pwdChangedTime && (count($pwdChangedTime) > 0)) {
-				if($pwdMaxAge && (count($pwdMaxAge) > 0)
-					&& $pwdExpireWarning && (count($pwdExpireWarning) > 0)) {
+			if((count($pwdChangedTime) > 0)) {
+				if((count($pwdMaxAge) > 0)
+					&& (count($pwdExpireWarning) > 0)) {
```
Comment (nickvergessen) [inline apps/user_ldap/lib/User/User.php]:
```suggestion
			if(!empty($pwdChangedTime)) {
				if(!empty($pwdMaxAge)
					&& !empty($pwdExpireWarning)) {
```

File: apps/user_ldap/lib/User/User.php [hunk 2]
```diff
@@ -752,16 +751,16 @@ public function handlePasswordExpiry($params) {
 				exit();
 			}
 			//handle pwdReset attribute
-			if($pwdReset && (count($pwdReset) > 0) && $pwdReset[0] === 'TRUE') { //user must change his password
+			if((count($pwdReset) > 0) && $pwdReset[0] === 'TRUE') { //user must change his password
```
Comment (nickvergessen) [inline apps/user_ldap/lib/User/User.php]:
```suggestion
			if (!empty($pwdReset) && $pwdReset[0] === 'TRUE') { //user must change his password
```

File: apps/user_ldap/lib/User/User.php [hunk 3]
```diff
@@ -732,15 +732,14 @@ public function handlePasswordExpiry($params) {
...
+			$pwdMaxAge = array_key_exists('pwdmaxage', $result[0]) ? $result[0]['pwdmaxage'] : [];
+			$pwdExpireWarning = array_key_exists('pwdexpirewarning', $result[0]) ? $result[0]['pwdexpirewarning'] : [];
 			
 			//handle grace login
 			$pwdGraceUseTimeCount = count($pwdGraceUseTime);
-			if($pwdGraceUseTime && $pwdGraceUseTimeCount > 0) { //was this a grace login?
-				if($pwdGraceAuthNLimit 
-					&& (count($pwdGraceAuthNLimit) > 0)
+			if($pwdGraceUseTimeCount > 0) { //was this a grace login?
+				if((count($pwdGraceAuthNLimit) > 0)
```
Comment (nickvergessen) [inline apps/user_ldap/lib/User/User.php]:
```suggestion
			if ($pwdGraceUseTimeCount > 0) { //was this a grace login?
				if (!empty($pwdGraceAuthNLimit))
```

**Review by nickvergessen — CHANGES_REQUESTED** [review]
If you only count to see you have more than 0 items, use `!empty()`

**Comment by lippl** [comment]
> @lippl Is this related to any existing issue?

I have not found any open issue for this yet but rather decided to go straight for a fix

> If you only count to see you have more than 0 items, use `!empty()`

@nickvergessen Thanks for the hint! I applied your suggested changes and also extended this to the initial  `count($pwdGraceUseTime)` for a variable by removing the variable, checking for `!empty` first and only later in case it is true use the `count()`, as this should be rare cases.
As you already started some pretty printing I also removed some unnecessary blank spaces as well.

> @lippl looks good from my point of view, just perhaps you can stash the commits into one?

If that is all fine with you, I would squash everything into one commit.

**Review by blizzz — APPROVED** [review]

**Comment by blizzz** [comment]
> If that is all fine with you, I would squash everything into one commit.

fine with me :)

**Review by juliusknorr — APPROVED** [review]

**Review by nickvergessen — APPROVED** [review]

**Comment by blizzz** [comment]
Thanks @lippl

**Pull request merged**